Code/Editor/EditorFramework/CodeGen/CppProject.h

```diff
@@ -2,10 +2,92 @@
 
 #include <EditorFramework/CodeGen/CppSettings.h>
 #include <EditorFramework/EditorFrameworkDLL.h>
+#include <EditorFramework/Preferences/Preferences.h>
 #include <Foundation/Communication/Event.h>
+#include <Foundation/Types/VariantType.h>
+#include <Foundation/Types/Status.h>
 
-struct EZ_EDITORFRAMEWORK_DLL ezCppProject
+// Only saved in editor preferences, does not have to work cross-platform
+struct EZ_EDITORFRAMEWORK_DLL ezIDE
 {
+  using StorageType = ezUInt8;
+
+  enum Enum
+  {
+    VisualStudioCode,
+#if EZ_ENABLED(EZ_PLATFORM_WINDOWS)
+    VisualStudio,
+#endif
+
+#if EZ_ENABLED(EZ_PLATFORM_WINDOWS)
+    Default = VisualStudio
+#else
+    Default = VisualStudioCode
+#endif
+  };
+};
+
+EZ_DECLARE_REFLECTABLE_TYPE(EZ_EDITORFRAMEWORK_DLL, ezIDE);
+
+// Only saved in editor preferences, does not have to work cross-platform
+struct EZ_EDITORFRAMEWORK_DLL ezCompiler
+{
+  using StorageType = ezUInt8;
+
+  enum Enum
+  {
+    Clang,
+#if EZ_ENABLED(EZ_PLATFORM_LINUX)
+    Gcc,
+#elif EZ_ENABLED(EZ_PLATFORM_WINDOWS)
+    Vs2022,
+#endif
+
+#if EZ_ENABLED(EZ_PLATFORM_WINDOWS)
+    Default = Vs2022
+#else
+    Default = Gcc
+#endif
+  };
+};
+
+EZ_DECLARE_REFLECTABLE_TYPE(EZ_EDITORFRAMEWORK_DLL, ezCompiler);
+
+struct EZ_EDITORFRAMEWORK_DLL ezCompilerPreferences
+{
+  ezEnum<ezCompiler> m_eCompiler;
+  bool m_bCustomCompiler;
+  ezString m_CppCompiler;
+  ezString m_CCompiler;
+  ezString m_RcCompiler;
+};
+
+EZ_DECLARE_REFLECTABLE_TYPE(EZ_EDITORFRAMEWORK_DLL, ezCompilerPreferences);
+
+struct EZ_EDITORFRAMEWORK_DLL ezCppProject : public ezPreferences
+{
+  EZ_ADD_DYNAMIC_REFLECTION(ezCppProject, ezPreferences);
+
+  struct MachineSpecificCompilerPaths
+  {
+    ezString sNiceName;
+    ezEnum<ezCompiler> eCompiler;
+    ezString sCCompiler;
+    ezString sCppCompiler;
+    bool bIsCustom;
```

Comment:
Not typical ez coding style.

Reply:
What exactly do you mean? We discussed member naming rules in depth, and public members don't need to start with `m_` we also do this in other places.

Reply:
I guess you are right, though I thought we only allow this because it would be idiotic for things like vec3. For more complex types we typically use m_. But I don't care that much TBH.

Reply:
No strong feelings either, added `m_`